### PR TITLE
Update getNumberOfComponents to return an int.

### PR DESCRIPTION
**Pull request by endJunction**
Closes #1995

### CODE REVIEW

File: Tests/NumLib/LocalToGlobalIndexMap.cpp [hunk 1]
```diff
@@ -163,7 +163,7 @@ TEST_F(NumLibLocalToGlobalIndexMapTest, DISABLED_MultipleVariablesMultipleCompon
         NumLib::ComponentOrder::BY_COMPONENT);
 
     ASSERT_EQ(30, dof_map->dofSizeWithGhosts());
-    ASSERT_EQ(3u, dof_map->getNumberOfComponents());
+    ASSERT_EQ(3, dof_map->getNumberOfComponents());
     ASSERT_EQ(2u, dof_map->getNumberOfVariables());
```
Comment (wenqing) [inline Tests/NumLib/LocalToGlobalIndexMap.cpp]: dof_map->getNumberOfVariables() also returns int. Therefore, 2u --> 2.

Comment (endJunction) [reply Tests/NumLib/LocalToGlobalIndexMap.cpp]: Thanks! I found few more places... Unfortunately I was not able to find/create an automatic substitution.

**Comment by wenqing** [comment]
@endJunction New warning in `assert(component < _n_components);` in  PropertyVector.h because that the type of  _n_components is changed to `int`, e.g. in line
https://github.com/ufz/ogs/blob/master/MeshLib/PropertyVector.h#L71

**Comment by endJunction** [comment]
Ah, I see. Unfortunately jenkins didn't show it. I'll fix it first thing in the morning.

@bilke We need a warning warning system ;)

**Pull request merged**